LibreHardwareMonitorLib/Hardware/Memory/MemoryGroup.cs

```diff
@@ -45,27 +35,7 @@ public MemoryGroup(ISettings settings)
         _hardware.Add(new TotalMemory(settings));
 
         //No RAM detected
-        if (!DetectThermalSensors(out List<SPDAccessor> accessors))
-        {
-            //Retry a couple of times
-            //SMBus might not be detected right after boot
-            _timer = new Timer(RetryTime);
```

Comment:
I don't think we can fully throw this away:
https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/pull/1743#discussion_r2219460141

Reply:
Well it can't be async, especially in the ctor flow of Computer/Hardware.

Reply:
The same issue exists in NetworkGroup, it's also possible to a network interface is removed/added there while enumerating. I think moving the retry here to Update might be an option, but there might still be some issues if _hardware is accessed during Update. I'll see if I can address NetworkGroup later.

Reply:
So either the retry period is short and it is blocking, or we need a fancier solution.

Reply:
Yeah, see as an example: https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/commit/8dc7d17e45e754a8e38256087190d56b8f7c2c70
I think we can do something similar with MemoryGroup.

Reply:
Updated.  Tried to make it as safe as possible so it doesn't crash if something goes wrong. Used task instead of timer to for cancellation token.